controllers/api/campaign.go

```diff
@@ -41,8 +42,28 @@ func (as *Server) Campaigns(w http.ResponseWriter, r *http.Request) {
 		if c.Status == models.CampaignInProgress {
 			go as.worker.LaunchCampaign(c)
 		}
-		JSONResponse(w, c, http.StatusCreated)
-	}
+
+    //TODO
```

Comment:
Just as a heads up, I'd recommend we move the logic to send a webhook event to `models/campaign.go` in `AddEvent`, since this is what is called anytime a campaign event happens.

Reply:
I wanted to do that initially. 

An issue would be that `AddEvent` does not only what it's supposed to do, but an additional thing. Which would bring issues: 

a) in testing it

b) in situations when we need only `AddEvent` and not (`AddEvent` + send webhook)

Therefore, I'd either a) create `AddEventEx` (AddEvent + webhook) b) call "sendWebhook" in the places where `AddEvent` and where we need a webhook to be sent.

Reply:
On second thought, I agree with what you're saying. I think part of the problem is that there is no way right now for `AddEvent` to know which webhook sender to use, requiring us to change the function signature in a very non-ideal way.

Instead, let's just go the manual route and add the `api.Webhook.Send` (using the `Webhook` property introduced in https://github.com/gophish/gophish/pull/1642#discussion_r344469030) wherever we add campaign events. If we want to reduce the amount of code repetition, we could just add a new private method off the API like `api.addCampaignEvent(campaign)` that runs both `AddEvent` and `api.Webhook.Send`, adjusting existing references accordingly.